makefiles/suit.v4.inc.mk

```diff
@@ -16,18 +16,46 @@ export CFLAGS += -DSOCK_URLPATH_MAXLEN=128
 SUIT_VENDOR ?= "riot-os.org"
 SUIT_SEQNR ?= $(APP_VER)
 SUIT_DEVICE_ID ?= $(BOARD)
-SUIT_KEY ?= secret.key
-SUIT_PUB ?= public.key
-SUIT_PUB_HDR ?= public_key.h
 
-$(SUIT_KEY) $(SUIT_PUB):
-	@$(RIOTBASE)/dist/tools/suit_v4/gen_key.py
+#
+# SUIT encryption keys
+#
+
+# Specify key to use.
+# Will use $(SUIT_KEY_DIR)/$(SUIT_KEY) $(SUIT_KEY_DIR)/$(SUIT_KEY).pub as
+# private/public key files, similar to how ssh names its key files.
+SUIT_KEY ?= default
+
+ifeq (1, $(RIOT_CI_BUILD))
+  SUIT_KEY_DIR ?= $(BINDIR)
+else
+  SUIT_KEY_DIR ?= $(RIOTBASE)/keys
+endif
+
+SUIT_SEC ?= $(SUIT_KEY_DIR)/$(SUIT_KEY)
+SUIT_PUB ?= $(SUIT_KEY_DIR)/$(SUIT_KEY).pub
+
+SUIT_PUB_HDR = $(BINDIR)/riotbuild/public_key.h
+SUIT_PUB_HDR_DIR = $(dir $(SUIT_PUB_HDR))
+CFLAGS += -I$(SUIT_PUB_HDR_DIR)
+BUILDDEPS += $(SUIT_PUB_HDR)
+
+$(SUIT_SEC) $(SUIT_PUB):
+	@echo suit: generating key pair in $(SUIT_KEY_DIR)
+	@mkdir -p $(SUIT_KEY_DIR)
+	@$(RIOTBASE)/dist/tools/suit_v4/gen_key.py $(SUIT_SEC) $(SUIT_PUB)
 
 $(SUIT_PUB_HDR): $(SUIT_PUB)
```

Comment:
I only have one comment here, if  `SUIT_PUB` already exists then the header file is never changed even if I override the key from `default_a` to `default_b`  then the header file is not changed, you would have to `make clean` before.

A solution could be adding `.PHONY: $(SUIT_PUB_HDR)` before so if I have 2 keys `default_a` `default_b` whenever I override `SUIT_KEY` from one option to another the header file is always changed, but this means all files depending on the header will always be compiled. (not sure if this is good, and I might mistaken about this)

Anyway as I said I think we should move forward, so if you have a valid argument for not doing this just move forward :)

Reply:
yup, indeed. I pushed a commit using FORCE and lazy-sponge. The former ensures re-building, the latter only rewrites the file if it actually changed. That should do it.